dhis-2/dhis-services/dhis-service-validation/src/main/java/org/hisp/dhis/validation/DefaultValidationResultService.java

```diff
@@ -120,4 +133,34 @@ public long countValidationResults( ValidationResultQuery query )
         List<Period> persistedPeriods = periodService.reloadPeriods( new ArrayList<>( periods ) );
         return validationResultStore.getValidationResults( orgUnit, includeOrgUnitDescendants, validationRules, persistedPeriods );
     }
+
+    private void validate( ValidationResultQuery query )
+    {
+        // check ou and vr filters to be valid UIDs
+        validateElements( query.getOu(), ErrorCode.E7500, ErrorMessage::new, CodeGenerator::isValidUid );
```

Comment:
Even if this is a reasonable check, we normally don't check whether the incoming UID is a valid UID or if the period ID is valid itself. We rather just check whether the metadata object actually exists in the database. So just for the sake of consistency, we could remove these checks.